plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftLaunchController.java

```diff
@@ -118,6 +116,7 @@ public void launch(ILaunchConfiguration configuration, String mode,
         launch.addProcess(new ServerProcess(launch, server, getLabel(launch.getLaunchMode())));
 		beh.setServerStarting();
 		waitForDeploymentConfigAndPods(server, monitor);
+		enableDevModeForNodeJsProject(OpenShiftServerUtils.getDeploymentConfig(server), server);
```

Comment:
turns out that the code that I added back into OpenShiftServerUtils#getDeploymentConfig(server) is identical to the currently existing OpenShiftServerUtils#getReplicationController(server). I thus simply did the change as suggested.